File: CPlusPlus/src/Owl.cpp
#include <iostream>
#include <string>
#include "../include/Owl.h"

namespace ClassZoo {
ClassZoo::Owl::Owl(std::string inputName, int inputID,
    float inputWeight, float inputHeight)
    : ClassZoo::Animal(inputName, inputID, inputWeight, inputHeight) {}
void ClassZoo::Owl::PrintSpecies() {
    std::cout << "Animal Species: Owl\n";
}
void ClassZoo::Owl::AnimalSound() {
    std::cout << ClassZoo::Animal::GetName() << " hoots\n";
}
}

File: Java/src/classZoo/Owl.java
package classZoo;

public class Owl extends Animal {

	public Owl(String inputName, int inputID, float inputWeight, float inputHeight) {
		super(inputName, inputID, inputWeight, inputHeight);
	}
	
	public void printSpecies() {
		System.out.println("Animal Species: Owl");
	}
	
	public void animalSound() {
		System.out.printf("%s hoots\n", this.getName());
	}

}
